src/Repositories/TodoRepository.ts

```diff
@@ -13,79 +11,152 @@ export interface Todo {
 
 //interface to implement repository pattern
 export interface RepositoryPI<T> {
-    add(objectToAdd: T);
+    add(todoToAdd: T);
     getAll();
-    getById(objectID: string);
+    getById(todoID: string);
     getByCompletionStatus(completedStatus: boolean);
-    remove(objectID: string);
-    update(objectID: string, objectToChange: T);
+    remove(todoID: string);
+    update(todoID: string, todoToChange: T);
 }
 
 /** 
  * The Todo Repository class implements all the standard repository pattern functions per the interface above
- * It will create a task array which contains the full list of todo tasks
  * should handle all the logic internally so the routes are abstracted to the basics of what it needs to know 
  * **/
-export class TodoRepository implements RepositoryPI<Todo> {
-    private taskArray: Todo[] = [];
-
+ export class TodoRepository implements RepositoryPI<Todo> {
+     /**
+      * This class has a series of functions called in the todoRoutes.
+      * Most will invoke the sequelize connection where they each have an 
+      * associated function that calls a query or multiple queries 
+      * these function will then return a todo or an array of all remaining todos
+      */
     constructor() {
     }
 
-    public add(task: Todo) {
-        //creates the task object then adds it to all tasks
-        this.taskArray.push(task)
-        return this.taskArray
+    async add(todoToAdd: Todo) {
+        /** adds each key to corresponding database columns. 
+         * Complete status is always 0 upon creation
+         * indicating someone would NOT make a todo if it was already complete
+         **/
+        const [results, metadata] = await instance.dbs.todoDB.sequelize.query(
+            `
+            INSERT INTO 
+                Todos 
+            VALUES (
+                "${todoToAdd.id}", 
+                "${todoToAdd.createdAt}", 
+                "${todoToAdd.name}", 
+                "${todoToAdd.dueDate}", 
+                "0"
+            )`)
```

Comment:
I left comments about this on Chase's PR as well.

Allowing user input directly into SQL queries is very dangerous and makes you vulnerable to a [SQL injection](https://www.w3schools.com/sql/sql_injection.asp) attack.

You can get around this vulnerability by [sanitizing](https://www.quora.com/What-exactly-is-data-sanitization-with-respect-to-SQL-injection) user input in all cases.

From my reading the best practice way to do this with Hapi is through either [replacements](https://sequelize.org/v7/manual/raw-queries.html#replacements) or [bind parameters](https://sequelize.org/v7/manual/raw-queries.html#bind-parameter)

Reply:
Fully preventing SQL injection attacks are also out of scope for this training this is primarily meant as an FYI in the future